src/main/kotlin/com/bardsoftware/papeeria/backend/Output.kt

```diff
@@ -26,54 +27,61 @@ import java.util.concurrent.Future
 import java.util.concurrent.atomic.AtomicInteger
 
 abstract class BaseOutput {
+  private val streamCounter = AtomicInteger(0)
```

Comment:
zero?

Reply:
It is incremented with every copyStream call but this could result in double call of close() (if first copyStream() and its coroutine completed before second copyStream() call). I refactored this code more, and moved more code to BaseOutput